styleguide/components/on-this-page/on-this-page.stories.js

```diff
@@ -0,0 +1,24 @@
+import initOnThisPage from '../../../src/ts/on-this-page/on-this-page';
+
+import docs from './on-this-page.mdx';
+import templateExample from '../../examples/on_this_page/example.html';
+import templateWithNestedLinks from '../../examples/on_this_page/with_nested_links.html';
+
+function decorateWithBehaviour(Story) {
+  setTimeout(() => initOnThisPage());
+  return Story();
+}
+
+export default {
+  title: 'Components/On this page',
+  parameters: { docs: { page: docs } },
+};
+
+export const Example = () => templateExample;
+Example.parameters = { docs: { source: { code: templateExample } } };
+
+export const WithNestedLinks = () => templateWithNestedLinks;
+WithNestedLinks.parameters = {
+  docs: { source: { code: templateWithNestedLinks } },
+};
+WithNestedLinks.decorators = [decorateWithBehaviour];
```

Comment:
In this file I had to add the decorator only on `WithNestedLinks`. It was running the js on the `Example` as well which caused problems - the nested links wouldn't open as the toggle was getting confused.